mbtiles/src/mbtiles.rs

```diff
@@ -223,7 +277,114 @@ impl Mbtiles {
+                zoom_level AS zoom,
+                count( ) AS count,
+                min( length( tile_data ) ) / 1024.0 AS smallest,
+                max( length( tile_data ) ) / 1024.0 AS largest,
+                avg( length( tile_data ) ) / 1024.0 AS average,
+                min(tile_column) as min_tile_x,
+                min(tile_row) as min_tile_y,
+                max(tile_column) as max_tile_x,
+                max(tile_row) as max_tile_y 
+            FROM tiles
+            GROUP BY zoom_level"#
+        );
+        let page_size = page_size_query.fetch_one(&mut *conn).await?.page_size;
+        let mb_type_string = match self.detect_type(&mut *conn).await? {
+            MbtType::Flat => "flat",
+            MbtType::FlatWithHash => "flat with hash",
+            MbtType::Normalized { .. } => "normalized",
+        };
+        let level_rows = tile_infos_query.fetch_all(&mut *conn).await?;
+        let mut level_details: Vec<LevelDetail> = level_rows
+            .into_iter()
+            .map(|r| {
+                let zoom = r.zoom.unwrap() as u8;
+                let count = r.count as u32;
+                let tile_length: f32 = 40075016.7 / (2_u32.pow(zoom as u32)) as f32;
+
+                let smallest = r.smallest.unwrap_or(0.0) as f32;
+                let largest = r.largest.unwrap_or(0.0) as f32;
+                let average = r.average.unwrap_or(0.0) as f32;
+
+                let min_tile_x = r.min_tile_x.unwrap();
+                let min_tile_y = r.min_tile_y.unwrap();
+                let max_tile_x = r.max_tile_x.unwrap();
+                let max_tile_y = r.max_tile_y.unwrap();
+
+                let minx = -20037508.34 + min_tile_x as f32 * tile_length;
+                let miny = -20037508.34 + min_tile_y as f32 * tile_length;
+                let maxx = -20037508.34 + (max_tile_x as f32 + 1.0) * tile_length;
+                let maxy = -20037508.34 + (max_tile_y as f32 + 1.0) * tile_length;
+
+                let bbox: [f32; 4] = [minx, miny, maxx, maxy];
+
+                LevelDetail {
+                    zoom: format!("{zoom}"),
+                    count,
+                    smallest,
+                    largest,
+                    average,
+                    bounding_box: bbox,
+                }
+            })
+            .collect();
+        let details_of_all = LevelDetail {
+            zoom: "all".to_string(),
+            count: level_details.iter().map(|l| l.count).sum(),
+            smallest: level_details
+                .iter()
+                .map(|l| l.smallest)
+                .reduce(f32::min)
+                .unwrap(),
+            largest: level_details
+                .iter()
+                .map(|l| l.largest)
+                .reduce(f32::max)
+                .unwrap(),
+            average: level_details.iter().map(|l| l.average).sum::<f32>()
+                / level_details.len() as f32,
+            bounding_box: [
+                level_details
```

Comment:
Bounds strtuct supports `+=` and `+` operators, so you can use `.map.reduce` too